content/courses/onchain-development/intro-to-anchor.md

```diff
@@ -406,8 +462,25 @@ pub struct InstructionAccounts<'info> {
 pub struct AccountStruct {
     data: u64
 }
+
+// Space calculation for account the AccountStruct
+const DISCRIMINATOR: usize = 0;
+const DATA_SIZE: usize = 8;
+
+impl Space for  AccountStruct {
+    const INIT_SPACE: usize = DISCRIMINATOR + DATA_SIZE;
+}
 ```
 
+#### Key takeaways:
+
+- The whole program structure can be broadly divided into three parts:
+  1. Instruction Logic: where the instruction handlers are defined (i.e., the
+     `#[program]` module).
+  2. Instruction Contexts: where the accounts required for the instructions are
+     configured.
+  3. Account Types: where the data structure of the accounts are defined.
+
```

Comment:
```suggestion
- The whole program structure can be broadly divided into three parts:
  1. Account constraints: define the accounts required for the instructions, as well as rules to apply to them - e.g., whether they need to sign the transaction, if they should be created on demand, how addresses for PDAs, etc.  
  2. Instruction handlers: implement program logic, as functions inside the`#[program]` module.
  3. Accounts: define the format used for data accounts. 
```

Anchor uses the term 'constraints' so we do too https://www.anchor-lang.com/docs/the-accounts-struct#constraints

Reply:
Also reordered this - constraints, then function, then account structs since that's the order they get executed.

Reply:
sure, I'll incorporate all this in the docs